lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1837,6 +1837,42 @@ bool QuakeBridgeVisitor::VisitInitListExpr(clang::InitListExpr *x) {
     // Pass initialization list with one member as a Ref.
     return pushValue(last[0]);
   }
+
+  // Check if all values here are Integer or Float Type
+  bool isAllIntOrAllFloat = [&]() {
+    for (auto v : last)
+      if (!v.getType().isIntOrFloat())
+        return false;
+    return true;
+  }();
+
+  // If these are integers or floats, then let's allocate
+  // some memory and store them there.
+  if (isAllIntOrAllFloat) {
+    // Clear the types for the init expr
+    typeStack.clear();
```

Comment:
Don't do this. It blindly wipes out all the surrounding type context.

Reply:
I thought this was probably not right. But I did have to remove those types to get the rest of the walk working. I will update this to just remove the last N types from the stack, where N is the number of elements in the init list.

Reply:
Yeah. This is a bit of a rough spot. When we visit things with builtin types like `int` we can get that type pushed on the type stack in an optional way. If the type was `struct Foo`, nothing would get pushed. Are these types coming out of the init-list visit? That may not be cleaning the type stack up properly.

Reply:
Yes, if I have 3 `double` values in the initializer_list, I end up seeing and retaining 3 `f64` types in the type stack.

Reply:
Without removing those types, I get 
```bash
cudaq-quake: /workspaces/cuda-quantum/lib/Frontend/nvqpp/ConvertDecl.cpp:506: bool cudaq::details::QuakeBridgeVisitor::VisitVarDecl(clang::VarDecl*): Assertion `typeStack.size() == typeStackDepth + 1 && "more than 1 type was added for the variable"' failed.
```

Reply:
OK. We were only caring about initializer lists with qubit expressions in them. Qubits are an intercepted type, so that should just leave an value on the value stack for each element in the list.

Now we're adding initializer lists of builtin typed expressions. So when we visit one of those expressions, it will dive all the way down and push a type. That type *should* be popped off the stack for each expression that pushes the next value on the value stack.

Hope that makes sense.

Reply:
Cool, so the `popType` calls are required.